### DISCUSSION TITLE
Using devProxy with a WebSocket server

### DISCUSSION THREAD

**Discussion by chrisdc** [nitrojs/nitro#1010] [Q&A]
Hello, I've been attempting to use Nitro's devProxy option to connect to a WebSocket server, but it never gets past the connecting state (and times out if I have that option enabled), can anyone see if I'm doing something wrong:

```js
// nitro.config.js
import { defineNitroConfig } from "nitropack";

export default defineNitroConfig({
  devProxy: {
    "/ws": {
      target: "ws://localhost:8080",
      ws: true,
    },
  },
});
```

When my WebSocket server is running `let ws = new WebSocket("ws://localhost:8080")` will connect, but `let ws = new WebSocket("ws://localhost:3000/ws")` fails. If it's helpful I've created a simple reproduction of the issue (including a very basic WebSocket server) at https://github.com/chrisdc/nitro-ws-proxy-test

**Comment by rschmied** [comment]
Are you still interested in a solution to this?  I've been running into a similar issue with Nuxt and Nitro where my websockets would simply refuse to connect, get an ECONNRESET -- I have made some changes to server.ts and got it to work for me. yymv -- might eventually do a PR

**Comment by yochze** [comment]
@rschmied would love to get more info!

**Reply by rschmied** [reply]
Sorry for the late reply -- wondering if you use bun for this?  If so, then I think there's a bug in the bun websocket client that prevents this from working.  I found out the hard way that it works just fine (for me) with node instead of bun.

**Reply by mhornbacher** [reply]
On node it didn't work for me :(
